### DISCUSSION TITLE
Como prosseguir a carreira?

### DISCUSSION THREAD

**Discussion by felipe-heredia** [frontendbr/forum#1729] [Discussão]
### Estou bem indeciso em qual tecnologia seguir.

Atualmente, faço algumas cópias de códigos de outras pessoas, como por exemplo Semana OmniStack da Rocketseat, entendo os códigos sei como eles funcionam mas não sei realmente em qual framework seguir.

Tentei seguir no backend, mas me senti preso, não tinha muitas ideias do que fazer, do que desenvolver para ficar apenas ali no backend.
Parti ao frontend com ReactJS, então travei no CSS, descobri que não sabia nada de CSS, porém copiando códigos em pensava que sabia.
Essa semana então, parti para o Mobile com ReactNative, fiz uma "cópia" do Instagram e simplesmente amei o ReactNative.

### Estou simplesmente indeciso do que fazer.

Depois desse alguns meses passando por todas essas diversidades, eu não sei se foco em uma ou em todas. Amo todas, mas quando vou para o CSS travo, vejo que não sei nada de UX ou apenas de fazer um sitizinho bem básico responsivo.

Se você (sim, você mesmo), tiver alguma dica, alguma ajuda pro garotão iniciante de 15 anos, ajuda o papai aqui pra poder entender o que posso fazer (se sigo nas 3) ou como posso fazer pra aprender cada um de forma lenta, ou simplesmente ter ideias, pois é isso que me falta no NodeJS.

---

Com carinho, Felipe Suri <3.

**Comment by theandersonn** [comment]
@felipesuri Em primeiro lugar e acima de tudo, parabéns pelo conhecimento que você já tem! Esse é um reflexo claro de seu esforço. 🏆

Vez por outra é normal nos sentirmos nessa bifurcação. 

Nas questões que você levantou, você também já apontou a solução, exemplo quando você falou sobre o CSS.

Estude a base!

- HTML semâmtico;
- As propriedades do CSS;
- CSS posicionamento e diagramação, sem framework;
- Automatizadores, como montar tasks do zero no Gulp e webpack;
- Processadores de CSS, SASS / Styles;
- JavaScript puro.

Pratique, veja aqui o que as empresas pedem no dia a dia. 👈
https://github.com/felipefialho/frontend-challenges

Documente seu aprendizado e avanço, já tem blog? Capricha no GitHub.

Esse esmero não vai passar despercebido.

Seja transparente com o que você sabe e o que não sabe, como fez aqui.

Ajude outras pessoas em fóruns. 😁

Sucesso em sua jornada.

**Comment by felipe-heredia** [comment]
**@theandersonn, em primeiro lugar obrigado pelo apoio e pelo reconhecimento.** 🔰

Agradeço pelas dicas no qual foi passada, irei me aprofundar em CSS daqui algum tempinho.

---
> Documente seu aprendizado e avanço, já tem blog? Capricha no GitHub.

No momento, ainda não tenho blog, mas cara, acabei vendo um [vídeo](https://www.youtube.com/watch?v=Ja6ZnJUQa_0) que me inspirou muito. O intuído principal do vídeo é lhe dizer o motivo de ter "arquivos" seus na internet. Eu já estava com pensamento em criar um blog, esse vídeo me inspirou ainda mais.

Nesses últimos tempos, estou arrumando meu GitHub, arrumar os README's e arrumar alguns projetos pra começar a criar uma imagem minha para outras pessoas/empresas.

Nas próximas semanas, vou terminar de organizar meu GitHub e começar o projeto do meu blog e começar a escrever sobre programação pra criar um nome **_"Felipe Suri"_** na internet.

---

Obrigado pela ajuda. 😉💖

**Comment by ninetails** [comment]
> alguma ajuda pro garotão iniciante de 15 anos

Meu, nem se preocupa... Vc tem tempo aí pra falhar 5 caminhos até encontrar seu rumo :laughing: 

Não encare **falha** como algo ruim, pejorativo. Encare falha como: quanto mais cedo vc falhar em algo, mais cedo vc vai encontrar algo que vc não curte ou que vc sabe da deficiência. E há formas para tratar delas.

Uma delas é escapar. Simplesmente vc pode escapar de algo que não curte e pegar outra carreira.

A outra é enfrentar. Analisar e atacar suas deficiências, ser masoquista e aprender. Não desistir e não mudar sua palavra pois é seu jeito ninja de ser.

Qual é a mais certa? É a **sua carreira**. Realmente não dá pra saber pra onde vc vai lá na frente sem ter estudado de tudo um pouco. Umas coisas vc vai fugir, outras vai enfrentar. Não há certo. Vai ter coisa que vc vai ter que aprender, querendo ou não, e isso também não é ruim (ex: se for seguir frontend, vc pode ~ser o pica em~ ~ser o cara foda em~ saber bastante js mas n adianta nada se ~for medíocre~ não ter muito conhecimento em css e exemplo assim vai ter em qualquer rumo tomado). Mas sempre vai ter algo que vc vai querer mais num certo momento e tudo vai ter seus _tradeoffs_.

E aí lá na frente o assunto vai virar: melhor ser _generalista_ ou _especialista_?

Faça o que você quiser, siga pessoas quem vc aspira, arranje um trampo que você seja feliz nele..

Vc tem 3 anos pra começar a escolher por onde começar... é tempo pra caramba. Se você já sabe que tem interesse em TI e já tá fazendo curso, digo o mesmo do @theandersonn : estude mais e pratique. Na prática vc vai entender onde vc tem mais afinidade e pra onde vc vai querer direcionar sua carreira. E se ainda não sabe para o que seguir, é pq ainda falta estudar e brincar mais com o que vc tá adquirindo de conhecimento. Vá além do que o curso te dá. Faça networking e arranje alguém pra te mentorar quando escolher um caminho :D

**Comment by willycamargo** [comment]
Primeiramente, parabéns pela curiosidade e esforço!

Eu comecei a estudar muito cedo também e não por escolha, mas por necessidade, com 15 anos eu já trabalhava na área, o que me fez ser efetivado como desenvolvedor frontend assim quando peguei a carteira de trabalho, ainda com 16 anos. Então a dica que vou dar aqui vai ser baseada na minha experiência no mercado desde a adolescência, e tirada totalmente da minha consciência com o que lembro das experiências que tive, sem ficar me aprofundando e nem linkando referências.

1. **Estude inglês.** Esse para mim é o ponto principal. Aproveite seu tempo livre e se aprofunde no idioma. A falta do inglês fluente me fez perder grandes oportunidades que eu gostaria de ter aproveitado e a dificuldade em consumir conteúdo no idioma me trouxe muita dificuldade para estudar teoria de algumas coisas, o que leva par ao próximo ponto;

2. **Estude a teoria.** Temos uma profissão que aprende muito na pática, e isso não é complementarmente ruim. Mas a falta de teoria pode atrapalhar bastante a sua compreensão e o avanço na carreira profissional. Por não ter tido a educação formal necessária, sofri muito em alguns projetos por não ter o domínio de alguns conceitos matemáticos, algoritmos, arquiteturas, linguagens etc...

3. **Não subestime a importância da experiência e das soft skills.**  Pode acontecer que o ego suba pra sua cabeça (e provavelmente acontece com a maioria dos programadores). Seja humilde diante de situações que exijam experiência e habilidades além de sua especialidade. Ser precoce em uma determinada área não te fará necessariamente bom em outra. Um exemplo claro de profissionais que amadurecem egocêntricos são inúmeros desenvolvedores que não entendem e não aceitam a forma como outros setores das organizações atuam, sempre apontando "erros" neles e "endeusando" o setor de T.I.

4. **Aprenda com os erros.** Assuma os seus erros e aprenda com eles. Não tem muito que falar aqui: desde situações mais técnicas e até em situações comportamentais, observe e aprenda. Quando um sistema sair do ar, quando um dado do banco for deletado etc. Tente racionalizar a origem do erro e tente não guardar magoas sobre isso, apenas aprendizados.

Agora, se todo esse planejamento não for útil, apenas arrume uma vaga de estágio assim que der, grude na cola de um mentor e vá embora aprendendo na prática. Foi como eu fiz, errei muito, tropecei muitas vezes, tive muita sorte e estou aqui, 8 anos depois ainda atuando na área :smile:

**Comment by MathCunhaDev** [comment]
Eae cara! 

**Primeiramente, parabéns pelo seu esforço até aqui!** 

Cara, assim como o @willycamargo eu comecei a estudar e a trabalhar no setor de T.I. bem cedo mas no início eu me dediquei ao suporte e infraestrutura e não a programação. Após alguns anos batendo cabeça e conhecendo muita gente eu vi que eu tinha uma vontade de construir coisas que as pessoas fossem usar e fossem legais para elas.

**1. Você vai mudar:** Você ainda vai amadurecer muito e muita coisa ainda vai acontecer na sua vida e elas vão te dar uma perspectiva diferente da pessoa que você é e, do que você quer. Por isso, primeiro viva, experimente coisas novas, faça um intercâmbio se puder e troque experiências, isso vai te dar novos pontos de vista. 

**2. Ideias:** Eu tinha (tenho) exatamente o mesmo problema que você de não ter ideias de saber o que fazer para me aprimorar. Minha solução foi começar a fazer todos os desafios de vagas que eu encontrava pela frente aqui no frontendbr e no trampos, todos os dias eu verificava os desafios e fazia os desafios das vagas que eu encontrava quando tinha um tempo (até que acabei sendo contratado por um delas hahaha).

**Links de ideias para praticar**
https://www.youtube.com/watch?v=H4CCPaYLTWg
https://programathor.com.br/blog/top-6-projetos-de-programacao-para-colocar-o-seu-conhecimento-em-pratica/
https://becode.com.br/aplicacoes-que-podem-ser-criadas-aprendendo-html-css-e-javascript/
https://github.com/felipefialho/frontend-challenges

**Comment by dionarodrigues** [comment]
@felipesuri creio que você já está munido de várias dicas e experiências excelentes que o pessoal compartilhou aqui. 

Como foi citado, **estude inglês**. Aprender inglês abrirá muitas portas: além da possibilidade de consumir conteúdos diretamente da fonte e escrito por pessoas do mundo todo, também trará muitas oportunidade internacionais, sejam elas de trabalho, projetos open source, networking...

E não se deixe intimidar pela quantidade de tecnologias, linguagens e ferramentas disponíveis para estudo. **Tenha calma, vá devagar e aprecie a jornada**. 😎

Creio que não há um único caminho correto para aquirir conhecimento, você precisará encontrar a sua maneira de alcançar uma carreira de sucesso como desenvolvedor, no entando, há muita inspiração e referências espalhadas por aí. Inclusive, criei um artigo recentemente sobre [Como se tornar um ótimo desenvolvedor front-end](https://www.diogorodrigues.dev/pt/blog/como-se-tornar-um-otimo-desenvolvedor-front-end). Confira lá e veja se o ajuda! ✌

Algumas dicas que acho ser essenciais:

- Foque no básico antes de partir para qualquer framework.
- Aprenda como a internet funciona e como ela interage com o browser
- Domine os fundamentos do HTML, CSS e JS
- Domine alguns princípios de design importantes para a criação de qualquer interface

E só então vá para conteúdos mais avançados. 

Veja uma lista completa de como se organizar entre os níveis iniciante, intermediário e avançado [**aqui**](https://www.diogorodrigues.dev/pt/blog/como-se-tornar-um-otimo-desenvolvedor-front-end).

Boa sorte! ✨
